rotkehlchen/chain/ethereum/graph.py

```diff
@@ -77,10 +84,42 @@ def query(
 
         querystr = prefix + querystr
         log.debug(f'Querying The Graph for {querystr}')
-        try:
-            result = self.client.execute(gql(querystr), variable_values=param_values)
-        except (requests.exceptions.RequestException, Exception) as e:
-            raise RemoteError(f'Failed to query the graph for {querystr} due to {str(e)}') from e
+
+        retries_left = QUERY_RETRY_TIMES
+        while retries_left > 0:
+            try:
+                result = self.client.execute(gql(querystr), variable_values=param_values)
+            except (requests.exceptions.RequestException, Exception) as e:
+                # NB: error status code expected to be in exception message
+                exc_msg = str(e)
+                is_retry_status_code = False
+                for status_code in RETRY_STATUS_CODES:
+                    if str(status_code.value) in exc_msg:
```

Comment:
Will every exception contain the status code? Why not query specific attribute of the exception? This is really error prone as it just checks for a number in a string.